### ISSUE TITLE
Immersive Engineering In-Game Manual Crashes Game

### ISSUE THREAD

**Issue by JaRyCu** [BluSunrize/ImmersiveEngineering#1689]
I'm playing through the FTB Presents Direwolf20 for 1.10 modpack that was released this week, and it seems as if the IE Manual crashes the client at random times.  I've had it happen 3x today, and it doesn't seem to generate any type of error when I look at the logs or the console.  I've seen another user report this on the Curseforge page with a little more detail, but my experience was solely with the machine the s/he referenced.

The most recent crash, roughly 10 minutes ago, was on a menu page.  I was reading the book and the game "blinked" and I was back at my desktop with the console window open.

_EDIT: After going back and looking at the main page for this, I went into my DW20 folder and looked at the logs, and there's no crash report folder.  If you can tell me where I might find this, I'd be more than happy to upload it for you to look through._

**Comment by Giovanni-NL** [comment]
Same here.  Using ImmersiveEngineering-0.10-48.jar

When viewing the page about the ear protection item I briefly see something like "Format error" or "Formatting error" text in the manual, and then the client crashes.  I do have crash log files however.  See the two attached files. 

#
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x00000000506d31fa, pid=5856, tid=18624
#
# JRE version: Java(TM) SE Runtime Environment (8.0_25-b18) (build 1.8.0_25-b18)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.25-b02 mixed mode windows-amd64 compressed oops)
# Problematic frame:
# V  [jvm.dll+0x4431fa]
#
# Failed to write core dump. Minidumps are not enabled by default on client versions of Windows
#
# An error report file with more information is saved as:
# C:\Users\R.D\Documents\Curse\Minecraft\Instances\GotCraft\hs_err_pid5856.log
#
# Compiler replay data is saved as:
# C:\Users\R.D\Documents\Curse\Minecraft\Instances\GotCraft\replay_pid5856.log
#
[hs_err_pid5856.log.txt](https://github.com/BluSunrize/ImmersiveEngineering/files/601337/hs_err_pid5856.log.txt)
[replay_pid5856.log.txt](https://github.com/BluSunrize/ImmersiveEngineering/files/601338/replay_pid5856.log.txt)

**Comment by Giovanni-NL** [comment]
![image](https://cloud.githubusercontent.com/assets/4085882/20454578/0c059960-ae45-11e6-84f4-6e5e61f73f62.png)

**Comment by BluSunrize** [comment]
Interesting! The ear defenders seem to be the most often offender, but I have little idea as to why...

Can you guys properly reproduce it? Not necessarily every time, but at a decent rate? Because that would make you great testcases for a fix =P

@malte0811 I personally have zero idea what causes the actual Java issues, but the format error was caused by the "%" sign at the end of the entry. I've fixed that one, so on the offchance that that was the cause for it, Next build might make this issue un-reproducable, aka fixed? ^^

**Comment by Giovanni-NL** [comment]
I'll keep flipping through the manual and let you guys know if/when this happens again.

**Comment by malte0811** [comment]
This is a duplicate of #1533. Try using up to date Java (Read https://github.com/BluSunrize/ImmersiveEngineering/issues/1533#issuecomment-256305849). If someone manages to reproduce this with recent Java I'll look into it more in-depth. I don't know why this is happening, but there are some things I can say (or guess) about the VM crash log:
- Some native code is trying to access address `0x0`. This is the C/C++ equivalent of a `NullPointerException`.
- There are some very interesting things in there (At least interesting to me. I haven't got any experience in analyzing VM crashes):
  - l. 297: `Lnet/minecraft/client/renderer/block/model/BakedQuG\FEџ^B\92@` (\FE and \92 seem to be unprintable characters): That was probably meant to be `BakedQuad`, but something seems to have corrupted that data.
  - l. 307: `D:\re\workspace\8-2-build-windows-amd64-cygwin\jdk8u25` and `built [...] with MS VC++ 10.0 (VS2010)`: Is this a JRE that was meant to be compiled with Cygwin/MinGW/something like that and was compiled with MS VisualC++ instead?

**Comment by mindforger** [comment]
@malte0811 this is the actual version packaged with minecraft .... i shed some tears of laugh to the fact they use such a cross compiled version ... as if they had compiled it themselfs ... but why?

**Comment by Giovanni-NL** [comment]
Hmm, interesting to see that the log shows that MC is running with this: 
`JRE version: Java(TM) SE Runtime Environment (8.0_25-b18) (build 1.8.0_25-b18) `
while I also have this installed: 
```
C:\Users\R.D>java -version
java version "1.8.0_111"
Java(TM) SE Runtime Environment (build 1.8.0_111-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.111-b14, mixed mode)
```
The runtime used is the one that Curse installs it seems, at C:\Users\R.D\Documents\Curse\Minecraft\Install\runtime\jre-x64\1.8.0_25\bin\javaw.exe

I'll update my profile to use build 111 instead and see if that solves the problem.

**Comment by malte0811** [comment]
I just installed Java 8 Update 25 (downloaded from Oracle) in a Windows VM and intentionally produced a JVM crash, it seems like the Cygwin/VC++ thing isn't caused by Mojang.

**Comment by mindforger** [comment]
the hell? let me guess, they changed it around update 56 or 65 .. as i can remember either one of these versions was a milestone for crashing modded minecraft ... i am still boggled why it is compiled this way

**Comment by BrisingrAerowing** [comment]
@mindforger A quick look at the OpenJDK code and it requires a Unix-like toolset (bash, sed, find, grep, etc.) for compilation on any platform. Run the configure script on Win32 and it will build with MSVC. Cygwin was likely just used to provide the needed tools.

Oracle Java is built on top of the OpenJDK, so I would presume the build system is the same.

It was likely done that way for the sake of simplicity, so that each platform didn't have to have its own build system that had to be kept in sync.

**Comment by snacdaws** [comment]
may i add that trying to look at the squeezer in the manual crashes it (also on Direwolf20 1.10.2 modpack)

also i am using the latest available jre

**Comment by malte0811** [comment]
I am pretty sure that that is a different issue that has been fixed (#1704). If it is generating a crash report in your crash-reports folder, it is something completely different from this one.
